### DISCUSSION TITLE
Loki: Suggest parser for log volume query so levels are shown in graph

### DISCUSSION THREAD

**Discussion by ivanahuckova** [grafana/grafana#41570] [Observability]
If we have log line in `logfmt` or `json` format and the log line has information about log level, we can use log parser to turn that level into a label and log volume can use this in visualisation. If the user doesn't use parser, Grafana could suggest to use it, as it could retrieve information about format and if it has level information. 
Without parser:

![image](https://user-images.githubusercontent.com/30407135/141180418-55ec0d09-4df8-4187-94f5-5fe57efe072d.png)

With parser:

![image](https://user-images.githubusercontent.com/30407135/141180518-64696f35-700d-4c21-96a2-e0950ff64c2e.png)

**Comment by ifrost** [comment]
We could also make the `level` field name configurable (it's hardcoded at the moment) as it's not super clear probably. Unless there's a way to map a field name? For example having a log line `message="hello" lvl="error"` is there a way to map `lvl` to `level` with `logfmt`?

**Reply by ivanahuckova** [reply]
Just fyi, we currently support following namings for level: `const labelNames = ['level', 'lvl', 'loglevel'];`. 

Yes, you can do the [line_format](https://grafana.com/docs/loki/latest/logql/log_queries/#line-format-expression) but that can become tricky for large/complex lines. Regarding configurable level field, we actually have feature request related to this https://github.com/grafana/grafana/issues/25585.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
